### ISSUE TITLE
Compile for windows

### ISSUE THREAD

**Issue by naaeef** [tpm2-software/tpm2-tss-engine#154]
Hello,
is it possible to compile this project for windows? I tried it with cygwin with no success.

Thank you for your help.

**Comment by AndreasFuchsSIT** [comment]
You will probably have to create a VS project and include all the files, I assume.
Sorry, but I'm not an expert here; i.e. no Windows compile box.

**Comment by paulduongemail** [comment]
@naaeef Any update?

**Comment by naaeef** [comment]
We managed to get the project running on windows on an experimental basis.
For compiling the tmp2-tss-engine we created a new Visual Studio Solution using Visual Studio 2019. In the solution, we created two projects with different build targets. One for the genkey Tool (for generating private keys in the TPM) with an executable file as a target and one with a DLL as a target.
Furthermore, we had to replace the function calls to sscanf with calls to sscanf_s. Also, we removed the header getopt.h and parsed the command line arguments manually.
In some header files, we also had to change the initialization of structs from {} to 0, because the syntax was not accepted by the compiler.

**Comment by AndreasFuchsSIT** [comment]
@naaeef Pretty cool. Would you mind turning this into a PR ?

**Comment by paulduongemail** [comment]
@naaeef Yes, if you'd like to create PR it will be very helpful and meaningful!
Thanks!

**Comment by naaeef** [comment]
I can't create a pull request because we did it on a very experimental base, and I currently do not have enough time to convert the code to a stable version.

**Comment by williamcroberts** [comment]
> I can't create a pull request because we did it on a very experimental base, and I currently do not have enough time to convert the code to a stable version.

Just upload it so its available, and then someone else can run with this.

**Comment by paulduongemail** [comment]
@naaeef Yes, please. Even though you just upload it, it could be a great reference for me. I was stuck for weeks.

**Comment by madaster97** [comment]
Wanted to pitch an alternative engine for windows for reference, [openssl-cng-engine](https://github.com/rticommunity/openssl-cng-engine). It uses the Windows Cryptography API: Next Generation (CNG)

I am planning to try and compile the tpm2 version on windows as well.
